csrc/logical_domain_map.cpp

```diff
@@ -368,6 +368,79 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     return dom_map;
   }
 
+  // TODO: refactor to use getNonMappingDomainInfo instead.
+  if (auto* op = dynamic_cast<GroupedMmaOp*>(consumer_tv_->definition())) {
+    bool has_g = TensorDomain::noReductions(consumer_root).size() == 3;
+    int64_t ndims_out = std::ssize(consumer_root);
+    // [rk] is the reduction axis for the matmul operation, it only exists if k
+    // is not broadcast.
+    bool has_rk = consumer_root.back()->isReduction();
+    int64_t out_non_rk_last_idx = has_rk ? ndims_out - 2 : ndims_out - 1;
+
+    int64_t last_producer_idx = std::ssize(producer_logical) - 1;
+    if (producer_tv_->sameAs(op->matrix1())) {
```

Comment:
```suggestion
    if (producer_tv_ == op->matrix1()) {
```
Question: what does sameAs buy us here? Isn't it guaranteed that `producer_tv_` is an input to `op`?

Reply:
Just out of habit. `sameAs` has a short-cut that checks if pointers are the same, so doesn't feel like a big issue either way.

Reply:
> Yeah I think pointer comparison is safe here

It isn't just safe. It must use the pointer comparison.